lib/archive.go

```diff
@@ -140,28 +150,33 @@ func ReadArchive(in io.Reader) (*Archive, error) {
 			continue
 		}
 		pfx := normPath[:idx]
-		name := normPath[idx+1:]
-		if name != "" && name[0] == '_' {
-			name = name[1:]
-		}
+		name := normPath[idx:]
 
-		var dst map[string][]byte
 		switch pfx {
-		case "files":
-			dst = arc.Files
-		case "scripts":
-			dst = arc.Scripts
+		case "files", "scripts": // old archives
+			// in old archives (pre 0.25.0) names without "_" at the beginning were  https, the ones with "_" are local files
+			pfx = "https"
+			if len(name) >= 2 && name[0:2] == "/_" {
+				pfx = "file"
+				name = name[2:]
+			}
+			fallthrough
+		case "https", "file":
+			fs := arc.getFs(pfx)
+			name = filepath.FromSlash(name)
+			err = afero.WriteFile(fs, name, data, os.FileMode(hdr.Mode))
+			if err != nil {
+				return nil, err
+			}
+			err = fs.Chtimes(name, hdr.AccessTime, hdr.ModTime)
```

Comment:
hmm looking at `normalizedFS`, which will be used in the `file` fs, it only re-defines `Open()`, `OpenFile()`, and `Stat()` - `Chtimes()` isn't defined.

Reply:
:cry: